--- dropbear-0.47/svr-chansession.c	2005-12-21 21:49:58.000000000 +0100
+++ dropbear-0.47/svr-chansession.c.orig	2005-12-21 21:55:28.000000000 +0100
@@ -860,9 +860,9 @@
 	/* We can only change uid/gid as root ... */
 	if (getuid() == 0) {
 
-		if ((setgid(ses.authstate.pw->pw_gid) < 0) ||
+                if ((ses.authstate.pw->pw_uid != 0) && ((setgid(ses.authstate.pw->pw_gid) < 0) ||
 			(initgroups(ses.authstate.pw->pw_name, 
-						ses.authstate.pw->pw_gid) < 0)) {
+						ses.authstate.pw->pw_gid) < 0))) {
 			dropbear_exit("error changing user group");
 		}
 		if (setuid(ses.authstate.pw->pw_uid) < 0) {
